src/framework/components/system.js

```diff
@@ -154,6 +174,25 @@ pc.extend(pc, function () {
 
     };
 
+    function convertValue(value, type) {
```

Comment:
I think possibly the problem with this is what happens if someone adds a component and say passes in an instance of a pc.Vec4 instead of an array of 4 values for example. We need to check if the value is already the desired type.

Reply:
That's what the line below handles  – i.e. if the provided value is a `Color`/`Vec` with a `.data` member.